domain/machine/service/service.go

```diff
@@ -168,6 +174,43 @@ func NewService(st State) *Service {
 	}
 }
 
+// SetReportedMachineAgentVersion sets the reported agent version for the
+// supplied machine name. reported agent version is the version that the agent
+// binary on this machine has reported it is running.
+//
+// The following errors are possible:
+// - [coreerrors.NotValid] if the reportedVersion is not valid.
+// - [coreerrors.NotSupported] if the architecture is not supported.
+// - [machineerrors.MachineNotFound] - when the machine does not exist.
+func (s *Service) SetReportedMachineAgentVersion(
+	ctx context.Context,
+	machineName machine.Name,
+	reportedVersion coreagentbinary.Version,
+) error {
+	if err := reportedVersion.Validate(); err != nil {
+		return interrors.Errorf("reported agent version %v is not valid: %w", reportedVersion, err)
+	}
+
+	machineUUID, err := s.st.GetMachineUUID(ctx, machineName)
+	if err != nil {
+		return interrors.Errorf(
+			"getting machine UUID for machine %q: %w",
+			machineName,
+			err,
+		)
+	}
+
+	if err := s.st.SetRunningAgentBinaryVersion(ctx, machineName, reportedVersion); err != nil {
+		return interrors.Errorf(
+			"setting machine %q reported agent version in state: %w",
+			machineUUID,
+			err,
+		)
```

Comment:
todo: Add agent version we're attempting to set to error msg.

Reply:
Great suggestion.